### PR TITLE
[Issue #578] Iimprove type definitions for search indexing.

### PR DESCRIPTION
**Pull request by LouisFettet**
Keeping the description bare here; this is an attempt to fix up type definitions by utilizing the generated definitions and performing some eldritch TS magic.

### CODE REVIEW

File: src/lib/components/ContentfulRichText/nodes/EntryHyperlink.svelte
Comment (LouisFettet) [inline src/lib/components/ContentfulRichText/nodes/EntryHyperlink.svelte:1]: Reverted without changes.

File: src/lib/components/SideNav/SideNav.server.ts
Comment (LouisFettet) [inline src/lib/components/SideNav/SideNav.server.ts:1]: Reverted without changes.

File: src/lib/loadEventsNewsMetadataMap.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
+import constructEventSlug from "./util/constructEventSlug";
 import type { ContentfulClient } from "$lib/services/server/contentful";
-import type { MetadataMapItem } from "./types";
+import type { PageMetadataMapItem } from "$lib/loadPageMetadataMap";
+
+import type { NewsAndEventsCollectionQuery } from "./$queries.generated";
 
 const queryNewsAndEvents = gql`
-  query NewsCollection {
+  query NewsAndEventsCollection {
```
Comment (LouisFettet) [inline src/lib/loadEventsNewsMetadataMap.ts:10]: We were missing the new meta fields below in the query. We should also be using `eventSummary` over `eventDescription`.

File: src/lib/loadEventsNewsMetadataMap.ts [hunk 2]
```diff
@@ -25,48 +30,35 @@ const queryNewsAndEvents = gql`
...
+
+type AdditionalMetadataFields = { url?: string | null; children?: string[] };
 
-// TODO: remove
-const constructEventSlug = (date: Date, slug: string): string =>
-  `${date.toISOString().split("T")[0]}-${slug}`;
+export type SearchIndexingMetadataMapItem =
+  | PageMetadataMapItem
+  | (NewsSearchIndexingMetadataMapItem & AdditionalMetadataFields)
+  | (EventSearchIndexingMetadataMapItem & AdditionalMetadataFields);
```
Comment (LouisFettet) [inline src/lib/loadEventsNewsMetadataMap.ts:54]: This isn't perfect, but it does give us what we need to ensure we're not querying for fields that don't exist. The downside here is that we need to check the `__typename` for the fields that don't overlap, but I think that's a fair trade-off and makes it easier to reason about.

File: src/lib/loadPageMetadataMap.ts
Comment (LouisFettet) [inline src/lib/loadPageMetadataMap.ts:1]: This file was reverted, with the only change being that I moved the `PageMetadataMapItemWithObjectID` definition into the Algolia status API route.

File: src/routes/+layout.svelte
Comment (LouisFettet) [inline src/routes/+layout.svelte:1]: Reverted without changes.

File: src/routes/+page.server.ts
Comment (LouisFettet) [inline src/routes/+page.server.ts:1]: Reverted without changes.

File: src/routes/api/v1/search-index/+server.ts [hunk 3]
```diff
@@ -16,6 +17,23 @@ const CONTENTFUL_ACTIONS = {
...
+    metaDescription = item?.metaDescription;
+  } else if (item?.__typename === "News") {
+    metaTitle = item?.metaTitle ?? item?.title;
+    metaDescription = item?.metaDescription ?? item?.subhead;
+  } else if (item?.__typename === "EventEntry") {
+    metaTitle = item?.metaTitle ?? item?.shortTitle;
+    metaDescription = item?.metaDescription ?? item?.eventSummary;
+  }
+  return { metaTitle, metaDescription };
+};
```
Comment (LouisFettet) [inline src/routes/api/v1/search-index/+server.ts:35]: This is a bit ugly, but we can be much more certain of the data we're looking at by switching on the `__typename`.

**Review by LouisFettet — COMMENTED** [review]
Self-review!

**Review by larsmaxwell — APPROVED** [review]

**Pull request merged**